cms/static/js/views/metadata.js

```diff
@@ -220,6 +225,44 @@ function(BaseView, _, MetadataModel, AbstractEditor, FileUpload, UploadDialog, V
 
     });
 
+    Metadata.NumberWithWarning = Metadata.Number.extend({
+        initialize: function() {
+            Metadata.Number.prototype.initialize.apply(this);
+            var options = this.model.get("options");
+            this.msg = options["warning"];
+            // show initial warning if necessary
+            this.checkWarning();
+        },
+
+        changed: function() {
+            Metadata.Number.prototype.changed.apply(this);
+            this.checkWarning();
+        },
+
+        checkWarning: function() {
+            var value = this.getValueFromEditor();
+            if (parseInt(value)) {
+                this.showWarning();
+            } else {
+                this.removeWarning(); 
+            }
+        },
+
+        showWarning: function() {
+            if (this.$(".setting-help-warning").length < 1) {
+                var $warning = this.$(".setting-help").clone();
+                $warning
+                    .text(this.msg)
+                    .addClass("setting-help-warning");
+                this.$(".setting-help").after($warning);
+            }
+        },
+
+        removeWarning: function() {
+            this.$(".setting-help-warning").remove();
```

Comment:
Instead of adding and removing elements from the DOM, could we instead either:
- hide/show the element
- remove/add the text